src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2818,7 +2818,13 @@ protected function castAttribute($key, $value)
             return $value;
         }
 
-        switch ($this->getCastType($key)) {
+        $castType = $this->getCastType($key);
+
+        if (method_exists($this, 'castTo' . strtoupper($castType))) {
```

Comment:
Might be worth to tweak that a bit:
```php
if (method_exists($this, $method = 'castTo'.ucfirst(Str::camel($castType)))) {
    return $this->$method($value);
}
```
So that `'foo_bar_baz'` type could be handled by `castToFooBarBaz($value)` ?